eldritch/mythos.py

```diff
@@ -412,6 +412,155 @@ def get_trigger(self, event, state):
+        and values.Calculation(event.gains["sanity"], operand=values.operator.gt, right=0).value(
+            state)
+        and (event.source is None or event.source.name not in ("Psychology", "Asylum"))
+    ):
+      return events.GainPrevention(self, event, "sanity", event.gains["sanity"])
+    return None
+
+
+class Mythos19(Headline):
+  def __init__(self):
+    super().__init__("Mythos19", "WitchHouse", "Cave", {"moon"}, {"plus"}, activity_location="Cave")
+    self.active_until = None
+
+  def create_event(self, state):
+    seq = super().create_event(state)
+    seq.events.extend([
+        events.CloseLocation("Merchant", for_turns=1, evict=False),
+        events.AddGlobalEffect(self, active_until=state.turn_number + 1, source_deck="mythos")
+    ])
+    return seq
+
+  def get_trigger(self, event, state):
+    if (
+        isinstance(event, events.Mythos)
+            and self.active_until is not None
+            and state.turn_number >= self.active_until):
+      return events.RemoveGlobalEffect(self, source_deck="mythos")
+    return None
+
+
+class Mythos20(Headline):
+  def __init__(self):
+    super().__init__("Mythos20", "Cave", "Roadhouse", {"moon"}, {"plus"})
+
+  def create_event(self, state):
+    seq = super().create_event(state)
+    seq.events.append(events.ReturnToCup(from_places=[
+        p.name for p in state.places if isinstance(p, places.Street)
+    ]))
+
+
+class Mythos21(Headline):
+  def __init__(self):
+    super().__init__("Mythos21", "Graveyard", "Isle", {"moon"}, {"plus"})
+
+  def create_event(self, state):
+    seq = super().create_event(state)
+    seq.events.append(events.IncreaseTerror())
+    return seq
+
+
+class Mythos22(Headline):
+  def __init__(self):
+    super().__init__("Mythos22", "Square", "Unnamable", {"hex"}, {"slash", "triangle", "star"})
+
+  def create_event(self, state):
+    seq = super().create_event(state)
+    cup_return = events.ReturnToCup(names=["Tentacle Tree"])
+    terrorize = events.Conditional(
+        None, cup_return, "returned", {0: events.Nothing(), 1: events.IncreaseTerror()}
+    )
+    seq.events.extend([cup_return, terrorize])
+    return seq
+
+
+class Mythos23(Headline):
+  def __init__(self):
+    super().__init__("Mythos23", "Unnamable", "Woods", {"moon"}, {"plus"})
+
+  def create_event(self, state):
+    seq = super().create_event(state)
+    seq.events.append(events.ReleaseMonstersToLocation("Merchant"))
+    return seq
+
+
+class Mythos24(Headline):
+  def __init__(self):
+    super().__init__("Mythos24", "WitchHouse", "Cave", {"circle"}, {"square", "diamond"})
+
+  def create_event(self, state):
+    seq = super().create_event(state)
+    seq.events.append(events.ReturnToCup(from_places=["Easttown", "Roadhouse", "Diner", "Police"]))
+
+
+class Mythos25(Environment):
+  def __init__(self):
+    super().__init__("Mythos25", "Woods", "Square", {"square", "diamond"}, {"circle"}, "weather")
+
+  def get_modifier(self, thing, attribute):
```

Comment:
This should only affect investigators in the city, not in other worlds. See Mythos6 for an example.